### ISSUE TITLE
Unreal 5.1 Kawaii Physics BUILD FAILED

### ISSUE THREAD

**Issue by malikeena** [pafuhana1213/KawaiiPhysics#62]
Hey guys I have been trying to upgrade Kawaii Physics to 5.1 for the past couple of weeks, had to change some classes because they seem to have changed their method for source objects  with TObjectPtr<>

	/** Settings for control of physical behavior */
	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Physics Settings", meta = (PinHiddenByDefault))
	FKawaiiPhysicsSettings PhysicsSettings;
	
	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> DampingCurve_DEPRECATED = nullptr;

	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> WorldDampingLocationCurve_DEPRECATED = nullptr;

	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> WorldDampingRotationCurve_DEPRECATED = nullptr;

	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> StiffnessCurve_DEPRECATED = nullptr;

	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> RadiusCurve_DEPRECATED = nullptr;

	/** Curve for adjusting the set value of physical behavior. Use rate of bone length from Root */
	UPROPERTY()
	TObjectPtr<UCurveFloat> LimitAngleCurve_DEPRECATED = nullptr;

Not sure if I did that right but it got rid of those issues for AnimNode_KawaiiPhysics.h

but still, have a couple more issues with something inside the editormoderegistry.h is still seeing it as an abstract class, Not sure how to fix this issue, but would like Kawaii Physics in 5.1 for our project.

I will keep debugging but here is how my last build went
I don't know how to rewrite FKawaiiPhysicseditmode so that it no longer is an abstract class

Severity	Code	Description	Project	File	Line	Suppression State
Error	C2259	'FKawaiiPhysicsEditMode': cannot instantiate abstract class	KawaiiUpgrade	X:\ProgramFiles\EpicGames\UnrealEngine-5.1\Engine\Source\Editor\UnrealEd\Public\EditorModeRegistry.h	125	

I'm stuck there if anyone could help I am new to c++, I know it has to do with calling the base

class FKawaiiPhysicsEditMode : public FKawaiiPhysicsEditModeBase
{	
public:
	FKawaiiPhysicsEditMode();
	/** IAnimNodeEditMode interface */
	virtual void EnterMode(class UAnimGraphNode_Base* InEditorNode, struct FAnimNode_Base* InRuntimeNode) override;
	virtual void ExitMode() override;
	virtual FVector GetWidgetLocation() const override;
	virtual UE_WIDGET::EWidgetMode GetWidgetMode() const override;
	virtual ECoordSystem GetWidgetCoordinateSystem() const override;
	virtual void DoTranslation(FVector& InTranslation) override;
	virtual void DoRotation(FRotator& InRotation) override;
	virtual void DoScale(FVector& InScale) override;

	/** FEdMode interface */
	virtual void Render(const FSceneView* View, FViewport* Viewport, FPrimitiveDrawInterface* PDI) override;
	virtual bool HandleClick(FEditorViewportClient* InViewportClient, HHitProxy* HitProxy, const FViewportClick& Click) override;
	virtual bool GetCustomDrawingCoordinateSystem(FMatrix& InMatrix, void* InData) override;
	virtual bool InputKey(FEditorViewportClient* InViewportClient, FViewport* InViewport, FKey InKey, EInputEvent InEvent) override;
	virtual bool ShouldDrawWidget() const override;
	virtual void DrawHUD(FEditorViewportClient* ViewportClient, FViewport* Viewport, const FSceneView* View, FCanvas* Canvas) override;

Can someone tell me how to rewrite this?

**Comment by XUJIANPENG1221** [comment]
Need to  Unreal 5.1 Kawaii Physics

**Comment by ZZwukong** [comment]
Need to Unreal 5.1 Kawaii Physics too......................XD

**Comment by franadoriv** [comment]
We really need KP on 5.1...

@malikeena Could you fork or make a branch with your advances please?

**Comment by i-Awoo** [comment]
i compiled it for 5.1 if anyone wants it
https://drive.google.com/file/d/11gtTg0y_HZU2Zp0mUu4Eo1Cbs5Arur0F/view?usp=sharing

**Comment by akyuu777** [comment]
> i compiled it for 5.1 if anyone wants it https://drive.google.com/file/d/11gtTg0y_HZU2Zp0mUu4Eo1Cbs5Arur0F/view?usp=sharing

amazing thank you

**Comment by ZZwukong** [comment]
> i compiled it for 5.1 if anyone wants it https://drive.google.com/file/d/11gtTg0y_HZU2Zp0mUu4Eo1Cbs5Arur0F/view?usp=sharing
> 
> 如果有人想要 https://drive.google.com/file/d/11gttg0y_hzu2zp0muu4eo1cbs5arur0f/view?usp=sharing 的话我花了5.1美元编辑的

You're amazing! Thank you very much. But still hope the author can continue to update, or I may have to give up the plug-in, in order to avoid future trouble...

**Comment by InsomniaGuy** [comment]
> i compiled it for 5.1 if anyone wants it https://drive.google.com/file/d/11gtTg0y_HZU2Zp0mUu4Eo1Cbs5Arur0F/view?usp=sharing
I used your vision,
but every time when I tweak value in spherial limits or capsule limits and compile, My UE 5.1 always crashes, do you have the same issue?

**Comment by akyuu777** [comment]
> > but every time when I tweak value in spherial limits or capsule limits and compile, My UE 5.1 always crashes, do you have the same issue?

this issue is present even in the past version for 5.0.X, use a KawaiiPhysicsLimitDataAsset to set the limits, and tweak from the data asset itself as well, refer to this guide if you need https://youtu.be/H4Gf6XMWh8I?t=974

this way is very practical for sharing the same data between nodes, but most important, its also stable

**Comment by muchogrande** [comment]
> i compiled it for 5.1 if anyone wants it https://drive.google.com/file/d/11gtTg0y_HZU2Zp0mUu4Eo1Cbs5Arur0F/view?usp=sharing

Thanks so much!

**Comment by blueroseslol** [comment]
I have upgraded this plugin to 5.1.
https://github.com/blueroseslol/KawaiiPhysics/releases/tag/20221121-v1.9.1

**Comment by InsomniaGuy** [comment]
> > > but every time when I tweak value in spherial limits or capsule limits and compile, My UE 5.1 always crashes, do you have the same issue?
> 
> this issue is present even in the past version for 5.0.X, use a KawaiiPhysicsLimitDataAsset to set the limits, and tweak from the data asset itself as well, refer to this guide if you need https://youtu.be/H4Gf6XMWh8I?t=974
> 
> this way is very practical for sharing the same data between nodes, but most important, its also stable

thanks dude!

**Comment by ZZwukong** [comment]
Thank you so much!

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年11月21日 20:49
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [pafuhana1213/KawaiiPhysics] Unreal 5.1 Kawaii Physics BUILD FAILED (Issue #62)


I have upgraded this plugin to 5.1.
https://github.com/blueroseslol/KawaiiPhysics/releases/tag/20221121-v1.9.1

―

**Comment by pafuhana1213** [comment]
Hi guys!
Thanks to you , I were able to support UE5.1!
https://github.com/pafuhana1213/KawaiiPhysics/commit/454f3de311438645e4f0af4e591f7a70fb1cc3c9

This update fixed some error and improve class design.

Updated for 5.1

Some classes moved to another module from UE5.1, so I added module name to build.cs
FAnimNodeEditMode was added Module API from UE5, so I used it as is in the UE5 version. However, in UE4 there is no API, so I continue to use the FAnimNodeEditMode clone KawaiiPhysicsEditModeBase. This class will be deleted at the same time when UE4 support is omitted
Fixed some warnings
Thanks!
